src/SessionSetup/ServiceDeployManager.cpp

```diff
@@ -300,20 +300,18 @@ outcome::result<void> ServiceDeployManager::CopyOrbitServicePackage() {
   return outcome::success();
 }
 
-ErrorMessageOr<void> ServiceDeployManager::CopyFileToLocal(std::string source,
-                                                           std::string destination) {
+orbit_base::Future<ErrorMessageOr<void>> ServiceDeployManager::CopyFileToLocal(
+    std::string source, std::string destination) {
   orbit_base::Promise<ErrorMessageOr<void>> promise;
   auto future = promise.GetFuture();
 
-  DeferToBackgroundThreadAndWait(
-      this, [this, source = std::move(source), destination = std::move(destination),
-             promise = std::move(promise)]() mutable {
-        promise.SetResult(CopyFileToLocalImpl(source, destination));
-      });
-
-  if (!future.IsFinished()) return ErrorMessage{"Copy operation was aborted."};
```

Comment:
Is this an error case that now needs to be handled elsewhere? Can't it happen now that OrbitApp::RetrieveModuleFromRemote returns a local path even though the download failed for some reason?

(That said, I don't fully understand the whole context in which the download mechanism is used.)